src/pages/RegistrationPage.tsx

```diff
@@ -39,19 +47,58 @@ const RegistrationPage: React.FC = () => {
     }
 
     if (!hasError) {
-      console.log('First Name: ', firstName);
-      console.log('Last Name: ', lastName);
-      console.log('Email: ', email);
-      console.log('Password: ', password);
+      const requestBody = {
+        email,
+        password,
+        firstName: firstName || 'FName',
+        lastName: lastName || 'LName',
+        gender: 'DECLINE_TO_IDENTIFY'
+      };
+      try {
+        const response = await fetch('/auth/register', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(requestBody)
+        });
+
+        if (response.ok) {
+          window.location.href = '/home';
+        } else {
+          const errorData = await response.json();
 
-      setFirstName('');
-      setLastName('');
-      setEmail('');
-      setPassword('');
-      setConfirmPassword('');
+          if (response.status === 406) {
+            setEmailFieldError('User with this email already exists.');
+          } else {
+            setServerError('Something went wrong, please try again!');
+            setIsModalOpen(true);
+          }
+          console.error('Failed to register', errorData);
+        }
+      } catch (error) {
+        console.error('An error occurred during registration', error);
+        setServerError('Something went wrong, please try again!');
+        setIsModalOpen(true);
+      }
     }
   };
 
+  const handleModalClose = () => {
+    setIsModalOpen(false);
+    setFirstName('');
+    setLastName('');
+    setEmail('');
+    setPassword('');
+    setConfirmPassword('');
+    setEmailFieldError(null);
+    setResetField(true);
```

Comment:
Ovaj flow nije bas standardan. Obicno ako user pogrijesi na nekoj formi, dobije neki vizualni indikator input field-a gdje je greska, i onda user samo ispravi to polje. Ovaj flow, u slucaju error-a, restartuje sva polja, i onda user mora sve ponovo ispuniti.

Reply:
Ovo se odnosi samo na zatvaranje modala, jer mi je u ticket-u naznaceno da se, ukoliko request bude neuspjesan (zbog server failure-a ili neceg sto je undefined), pojavi pop-up message na cije zatvaranje se sva polja trebaju isprazniti. Posebno je radjena logika za validaciju email-a i password-a koja radi po tom principu.